src/types/processor.ts

```diff
@@ -41,4 +42,5 @@ export type Processor = {
   initialCursor?: Cursor | undefined;
 };
 
-export type MapTrack = (nft: NFT, apiTrack: any, nftFactory?: NftFactory | undefined, trackId?: string) => ProcessedTrack;
+export type MapTrack = (nft: NFT, apiTrack: any, contract?: NftFactory, trackId?: string) => ProcessedTrack;
+export type MapNFTsToTrackIds = (nfts: NFT[], dbClient?: DBClient, apiTracksByNFT?: any, extractor?: Extractor) => { [trackId: string]: NFT[] };
```

Comment:
given we're refactoring these types now, i think it also makes sense to make the input be a object with each item as a field, rather than each item being a function parameter (similar to how MapArtistProfile is today)